.github/workflows/main.yml

```diff
@@ -14,3 +14,36 @@ jobs:
       - uses: actions/checkout@v2
       - name: Test
         run: ./test.sh
+  upload:
+    needs: build
+    if: startsWith(github.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    env:
+      name: omero-server
+    steps:
+      - name: Get prefix
+        id: getprefix
+        run: |
+          if [ ! -z ${{ env.name }} ]; then
+            echo "::set-output name=prefix::${{ github.actor }}/${{ env.name }}:"
+          else
+            echo "::set-output name=prefix::${{ github.repository }}:"
+          fi
+      - name: Get other tags
+        id: gettags
+        uses: jupyterhub/action-major-minor-tag-calculator@v1.1.0
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          prefix: "${{ steps.getprefix.outputs.prefix }}"
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v2
```

Comment:
As a follow-up, we could take advantage of the action `labels` property (see https://github.com/docker/build-push-action/blob/master/docs/advanced/tags-labels.md) and start generating the metadata that's currently hardcoded and updated via https://github.com/ome/omero-server-docker/blob/eba93585868447eb84f84d3582106cf4dae7d455/Makefile#L36-L38 as part of the build process